### ISSUE TITLE
systemd-cryptsetup fails to unlock with FIDO2 token 

### ISSUE THREAD

**Issue by 239** [systemd/systemd#19252]
**systemd version the issue has been seen with**
 > 248

**Used distribution**
 > Arch Linux with systemd 248-3

**Linux kernel version used** (`uname -a`)

 > `Linux U4 5.11.11-arch1-1 #1 SMP PREEMPT Tue, 30 Mar 2021 14:10:17 +0000 x86_64 GNU/Linux`

**CPU architecture issue was seen on**
 > x86-64

**Expected behaviour you didn't see**
 > Asking FIDO2 token (SoloKey) for PIN / button

**Unexpected behaviour you saw**
 > Failed to activate with FIDO2 decrypted key. (Key incorrect?)
 > Failed to extract keyslot index from FIDO2 JSON data: Wrong medium type

**Steps to reproduce the problem**
 > sudo systemd-cryptenroll --fido2-device=auto /dev/...
 > sudo /usr/lib/systemd/systemd-cryptsetup attach fido2test /dev/... - fido2-device=auto

**Additional program output to the terminal or log subsystem illustrating the issue**

```text
➤ sudo systemd-cryptenroll --fido2-device=auto /dev/sdb1
🔐 Please enter current passphrase for disk /dev/sdb1: ...                     
Initializing FIDO2 credential on security token.
👆 (Hint: This might require verification of user presence on security token.)
🔐 Please enter security token PIN: ...                    
Generating secret key on FIDO2 security token.
New FIDO2 token enrolled as key slot 1.

➤ sudo /usr/lib/systemd/systemd-cryptsetup attach fido2test /dev/sdb1 - fido2-device=auto
Set cipher aes, mode xts-plain64, key size 512 bits for device /dev/sdb1.
Automatically discovered security FIDO2 token unlocks volume.
Asking FIDO2 token for authentication.
Failed to activate with FIDO2 decrypted key. (Key incorrect?)
🔐 Please enter passphrase for disk Basic data partition (fido2test): (press TAB for no echo)
```

Probably a duplicate of #19246 but I'm not sure as I don't know how to get more details about the error.

**Comment by poettering** [comment]
Hmm, do you have a PIN set on the device? does it work if you drop it?

**Comment by 239** [comment]
Yes, PIN was set and without it works fine. I could not drop the PIN without resetting the SoloKey so I had to enroll again.
I also tested unlocking the root volume on boot and that also worked now but it didn't ask for confirmation (pressing button). Is that working as intended?

**Comment by poettering** [comment]
Let's close this, as this most likely got fixed by #19298. Please report back if this remains a problem with hat PR applied.
